hack/windows/Clean-AntreaNetwork.ps1

```diff
@@ -15,33 +15,119 @@ function GetHnsnetworkId($NetName) {
+        $OVS_DB_PATH = "$OVSInstallDir\etc\openvswitch\conf.db"
+        Remove-Item -Path "$OVSInstallDir\etc\openvswitch\*"
+        ovsdb-tool create "$OVS_DB_PATH" "$OVS_DB_SCHEMA_PATH"
+
+        start-service ovsdb-server
+        sc.exe create ovs-vswitchd binpath="$OVSInstallDir\usr\sbin\ovs-vswitchd.exe  --pidfile -vfile:info --log-file  --service" start= auto depend= "ovsdb-server"
+        sc.exe failure ovs-vswitchd reset= 0 actions= restart/0/restart/0/restart/0
+        start-service ovs-vswitchd
     }
-    if ($RetryCount -eq $MaxRetryCount) {
-        break
+}
+
+function RemoveNetworkAdapter($adapterName) {
+    $adapter = $(Get-NetAdapter $adapterName -ErrorAction Ignore)
+    if ($adapter -ne $null) {
+        Remove-NetIPAddress -IfAlias $adapterName -Confirm:$false
+        Write-Host "Network adapter $adapter.Name is left on the Windows host with status $adapter.Status, please remove it manually."
     }
-    Start-Sleep -Seconds 5
 }
-if (!$BrIntDeleted) {
-    Write-Host "Failed to delete OVS Bridge, please retry the script or delete the bridge and HNS network manually."
-    return
+
+function RemoveHiddenNetDevices() {
+    $Devs = $(Get-PnpDevice -Class net | ? Status -eq Unknown | Select InstanceId)
+    foreach ($Dev in $Devs) {
+        $RemoveKey = "HKLM:\SYSTEM\CurrentControlSet\Enum\$($Dev.InstanceId)"
+        Get-Item $RemoveKey | Select-Object -ExpandProperty Property | %{Remove-ItemProperty -Path $RemoveKey -Name $_ -Verbose }
+    }
 }
 
+function clearOVSBridge() {
+    $ovsStatus = $(Get-Service ovs-vswitchd).Status
+    if ("$ovsStatus" -EQ "running") {
+        Write-Host "Delete OVS bridge: br-int"
+        ovs-vsctl.exe --no-wait --if-exists del-br br-int
+        $MaxRetryCount = 10
+        $RetryCountRange = 1..$MaxRetryCount
+        $BrIntDeleted = $false
+        foreach ($RetryCount in $RetryCountRange) {
+            Write-Host "Waiting for OVS bridge deletion complete ($RetryCount/$MaxRetryCount)..."
+            $BrIntAdapter = $(Get-NetAdapter br-int -ErrorAction SilentlyContinue)
+            if ($BrIntAdapter -eq $null) {
+                $BrIntDeleted = $true
+                break
+            }
+            if ($RetryCount -eq $MaxRetryCount) {
+                break
+            }
+            Start-Sleep -Seconds 5
+        }
+        if (!$BrIntDeleted) {
+            Write-Host "Failed to delete OVS Bridge, please retry the script or delete the bridge and HNS network manually."
+            return
+        }
+    }
+}
 
+$BrIntDeleted = $(Get-NetAdapter br-int) -Eq $null
+if ($BrIntDeleted -eq $false) {
+    clearOVSBridge
+}
+$uplink = ""
+$AntreaHnsNetworkName = "antrea-hnsnetwork"
 $NetId = GetHnsnetworkId($AntreaHnsNetworkName)
 if ($NetId -ne $null) {
     Write-Host "Remove HnsNetwork: $AntreaHnsNetworkName"
+    $uplink = $(Get-HnsNetwork -Id $NetId).NetworkAdapterName
     Get-HnsNetwork -Id $NetId | Remove-HnsNetwork
 }
+
+ResetOVSService "ovs-vswitchd"
```

Comment:
Could add some comments to explain why reset?

Reply:
Done.